### DISCUSSION TITLE
'lte' as a key value in postman request

### DISCUSSION THREAD

**Discussion by Mazeral** [prisma/prisma#14257] [Q&A] [answered]
Hello, I've created a `remove` function the consists of the `delete` and the `findFirst` api, however strange thing happend, the find function threw an interesting error that does't make sense to me.
It says between the lines `did you mean lte?` which sounds strange but when I changed the key to `lte` the function worked.
Why does prisma accepts 'lte' as a key value in the request?
![Screenshot from 2022-07-12 10-39-51](https://user-images.githubusercontent.com/60851657/178460100-f5a93a72-17bb-4e98-accc-980c63fb9458.png)

![Screenshot from 2022-07-12 11-09-14](https://user-images.githubusercontent.com/60851657/178460107-17fb41cd-7184-42ce-bcdb-1c75d887765b.png)

my prisma shema looks like this :
```js
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Author {
  id  Int @id @default(autoincrement())
  FirstName String 
  LastName String
  books Book[]
}

model Book{
  id Int @id @default(autoincrement())
  Title String
  Description String
  authors Author[]
}
```
and the `remove` function looks like this :
```ts
  async remove(title: string) {
    try {
      const bookID: {
        id: number;
      } = await this.prisma.book.findFirst({
        where: {
          Title: title,
        },
        select: {
          id: true,
        },
      });
      await this.prisma.book.delete({
        where: {
          id: bookID.id,
        },
      });
      return { deleted: true };
    } catch (err) {
      return { deleted: false, message: err.message };
    }
  }```

**Comment by Mazeral** [comment]
I've managed to display the error in more clear way
```
Invalid `this.prisma.book.findFirst()` invocation in
/home/mohammadomar/training/library-dojo/src/book/book.service.ts: 93: 34

   90 try {
    91   const bookID: {
        92     id: number;
→  93
    } = await this.prisma.book.findFirst({
          where: {
            Title: {
              title: 'another book'
              ~~~~~
            }
        },
          select: {
            id: true
        }
    })

Unknown arg `title` in where.Title.title for type StringFilter. Did you mean `lte`? Available args:
type StringFilter {
  equals?: String
  in?: List<String>
  notIn?: List<String>
  lt?: String
  lte?: String
  gt?: String
  gte?: String
  contains?: String
  startsWith?: String
  endsWith?: String
  mode?: QueryMode
  not?: String | NestedStringFilter
    }
    ```

**Comment by austincrim** [comment, marked as answer]
Hey there!

It looks like the `title` variable is getting passed in as an object `{title: 'another book'}` instead of a plain string. Prisma is expecting either a string or a filter object like the one printed in the error message.

It is suggesting `lte` which is the `less than or equal` option because it is similar in spelling to `title` 🙂.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
